lucene/core/src/test/org/apache/lucene/index/TestDocumentsWriterDeleteQueue.java

```diff
@@ -36,6 +37,26 @@
  */
 public class TestDocumentsWriterDeleteQueue extends LuceneTestCase {
 
+
+  public void testAdvanceReferencesOriginal() {
+    WeakAndNext weakAndNext = new WeakAndNext();
+    DocumentsWriterDeleteQueue next = weakAndNext.next;
+    assertNotNull(next);
+    System.gc();
```

Comment:
this is terrible but I don't know of a better way of doing this. this at least works on my machine :) and would have found the issue

Reply:
@uschindler @dweiss maybe you have better ideas